### DISCUSSION TITLE
How to pass previous step output to writehttpresponse activity

### DISCUSSION THREAD

**Discussion by neerajmathur** [elsa-workflows/elsa-core#4419] [Q&A] [answered]
Hi,

I have just started playing around with ELSA, and wondering how to pass previous step output to WriteReponse activity?

In below workflow In and write a http response which is generated by RunJavaScript activity? I looked around documentation but unable to find answer, can you point me to a documentation

  builder
                .HttpEndpoint("/hello-world")
                .SetVariable("firstName", "Neeraj")
                .SetVariable("lastName", "Mathur")
                .RunJavaScript("`Hello ${firstName} ${lastName}`")
                .WriteHttpResponse(HttpStatusCode.OK, "`{{RunJavaScript.Output.Body}}`(Liquid)", "text/html");

**Comment by sfmskywalker** [comment, marked as answer]
Hello @neerajmathur , the following will work:

```csharp
builder
    .HttpEndpoint("/hello-world")
    .SetVariable("firstName", "Neeraj")
    .SetVariable("lastName", "Mathur")
    .RunJavaScript("`Hello ${firstName} ${lastName}!`")
    .WriteHttpResponse(_ => HttpStatusCode.OK, context => $"{context.Input!} (Liquid)", _ => "text/html");
```

Notice that the WriteHttpResponse extension method is using an overload that accept `ActivityExecutionContext` as an argument for each lambda argument. This allows you to access things like input, which contains the output of the last-executed activity.

**Comment by neerajmathur** [comment]
Thank You, This helped
